File: examples/04_application/stock/stock.js
// Lean Code — BSD 3-Clause License — Vivian Voss, 2026
// Scope: Read and check stock availability.

import { readFileSync } from "node:fs";

function readStockForProduct(product, stockPath) {
  const lines = readFileSync(stockPath, "utf-8").trim().split("\n");

  for (const line of lines) {
    const [name, count] = line.split(",", 2);
    if (name === product) {
      return parseInt(count, 10);
    }
  }

  return 0;
}

function checkStockAvailability(product, quantity, stockPath) {
  const stock = readStockForProduct(product, stockPath);
  return stock >= quantity;
}

export { readStockForProduct, checkStockAvailability };

File: examples/04_application/stock/stock.py
# Lean Code — BSD 3-Clause License — Vivian Voss, 2026
# Scope: Read and check stock availability.


def read_stock_for_product(product, stock_path):
    try:
        lines = open(stock_path).read().strip().split("\n")
    except FileNotFoundError:
        return 0

    for line in lines:
        name, count = line.split(",", 1)
        if name == product:
            return int(count)

    return 0


def check_stock_availability(product, quantity, stock_path):
    stock = read_stock_for_product(product, stock_path)
    return stock >= quantity
